R/module_teal.R

```diff
@@ -1,111 +1,138 @@
-#'   returns the `teal_data`, only evaluated once, `NULL` value is ignored
+#' @inheritParams module_init_data
+#' @inheritParams init
 #'
 #' @return
-#' Returns a `reactive` expression which returns the currently active module.
-#'
-#' @keywords internal
-#'
+#' `NULL` invisibly
 NULL
 
 #' @rdname module_teal
+#' @export
 ui_teal <- function(id,
-                    splash_ui = tags$h2("Starting the Teal App"),
+                    modules,
+                    data = NULL,
                     title = build_app_title(),
                     header = tags$p(),
                     footer = tags$p()) {
   checkmate::assert_character(id, max.len = 1, any.missing = FALSE)
-
-  checkmate::assert_multi_class(splash_ui, c("shiny.tag", "shiny.tag.list", "html"))
+  checkmate::assert_multi_class(data, "teal_data_module", null.ok = TRUE)
+  checkmate::assert(
+    .var.name = "title",
+    checkmate::check_string(title),
+    checkmate::check_multi_class(title, c("shiny.tag", "shiny.tag.list", "html"))
+  )
+  checkmate::assert(
+    .var.name = "header",
+    checkmate::check_string(header),
+    checkmate::check_multi_class(header, c("shiny.tag", "shiny.tag.list", "html"))
+  )
+  checkmate::assert(
+    .var.name = "footer",
+    checkmate::check_string(footer),
+    checkmate::check_multi_class(footer, c("shiny.tag", "shiny.tag.list", "html"))
+  )
 
   if (is.character(title)) {
     title <- build_app_title(title)
   } else {
     validate_app_title_tag(title)
   }
 
-  checkmate::assert(
-    .var.name = "header",
-    checkmate::check_string(header),
-    checkmate::check_multi_class(header, c("shiny.tag", "shiny.tag.list", "html"))
-  )
   if (checkmate::test_string(header)) {
     header <- tags$p(header)
   }
 
-  checkmate::assert(
-    .var.name = "footer",
-    checkmate::check_string(footer),
-    checkmate::check_multi_class(footer, c("shiny.tag", "shiny.tag.list", "html"))
-  )
   if (checkmate::test_string(footer)) {
     footer <- tags$p(footer)
   }
 
   ns <- NS(id)
 
-  # Once the data is loaded, we will remove this element and add the real teal UI instead
-  splash_ui <- tags$div(
-    # id so we can remove the splash screen once ready, which is the first child of this container
-    id = ns("main_ui_container"),
-    # we put it into a div, so it can easily be removed as a whole, also when it is a tagList (and not
-    # just the first item of the tagList)
-    tags$div(splash_ui)
-  )
-
   # show busy icon when `shiny` session is busy computing stuff
   # based on https://stackoverflow.com/questions/17325521/r-shiny-display-loading-message-while-function-is-running/22475216#22475216 # nolint: line_length.
   shiny_busy_message_panel <- conditionalPanel(
     condition = "(($('html').hasClass('shiny-busy')) && (document.getElementById('shiny-notification-panel') == null))", # nolint: line_length.
     tags$div(
-      icon("arrows-rotate", "spin fa-spin"),
+      icon("arrows-rotate", class = "fa-spin", prefer_type = "solid"),
```

Comment:
`fontawesome` package doesn't play nicely with extra class and explicit icon definition. [Issue created](https://github.com/rstudio/fontawesome/issues/122) upstream